apps/web/src/services/dashboards/pages/DashboardsMainPage.vue

```diff
@@ -119,14 +129,66 @@ const queryState = reactive({
 });
 
 /* Util */
-const getSearchedTreeData = (treeData: TreeNode<DashboardTreeDataType>[], searchFilters: ConsoleFilter[]): TreeNode<DashboardTreeDataType>[] => {
-    if (!searchFilters.length) return treeData;
-    return treeData.filter((node) => {
-        if (node.data.type === 'DASHBOARD') return true;
-        return !!node.children?.length;
+const getSearchedTreeData = (treeData: TreeNode<DashboardTreeDataType>[], searchedDashboardIdList: Set<string>): TreeNode<DashboardTreeDataType>[] => {
+    if (state.isSearching) return [];
+    if (!state.searchFilters.length) return treeData;
+    const _results = [] as TreeNode<DashboardTreeDataType>[];
+    treeData.forEach((node) => {
+        const _node = cloneDeep(node);
+        if (_node.data.type === 'DASHBOARD') {
+            if (searchedDashboardIdList.has(_node.data.id)) {
+                _results.push(_node);
+            }
+        } else {
+            const _children = getSearchedTreeData(_node.children || [], searchedDashboardIdList);
+            if (_children.length) {
+                _node.children = _children;
+                _results.push(_node);
+            }
+        }
     });
+    return _results;
+};
+const setSearchedDashboardIdList = async () => {
+    state.isSearching = true;
+    state.searchedDashboardIdList.clear();
+    let _publicDashboardIdList: string[] = [];
+    let _privateDashboardIdList: string[] = [];
+    if (storeState.isAdminMode) {
+        _publicDashboardIdList = await searchDashboards('PUBLIC');
+    } else if (storeState.isWorkspaceOwner) {
+        _publicDashboardIdList = await searchDashboards('PUBLIC');
+        _privateDashboardIdList = await searchDashboards('PRIVATE');
+    } else {
+        _publicDashboardIdList = await searchDashboards('PRIVATE');
+    }
+    _publicDashboardIdList.forEach((id) => state.searchedDashboardIdList.add(id));
+    _privateDashboardIdList.forEach((id) => state.searchedDashboardIdList.add(id));
+    state.isSearching = false;
 };
 
+/* Api */
+const searchApiQueryHelper = new ApiQueryHelper();
+const searchDashboards = async (dashboardType: 'PUBLIC' | 'PRIVATE'): Promise<string[]> => {
+    try {
+        searchApiQueryHelper.setFilters(state.searchFilters);
+        const fetcher = dashboardType === 'PRIVATE'
+            ? SpaceConnector.clientV2.dashboard.privateDashboard.list
+            : SpaceConnector.clientV2.dashboard.publicDashboard.list;
+        const { results } = await fetcher<DashboardListParams, ListResponse<DashboardModel>>({
+            query: {
+                only: ['dashboard_id'],
+                ...searchApiQueryHelper.data,
+            },
+        });
+        return results?.map((d) => d.dashboard_id) || [];
+    } catch (e) {
+        ErrorHandler.handleError(e);
+        return [];
+    }
+};
```

Comment:
issue: I think if the user search continuously, the previous response can be cancelled.
But `searchDashboards` returns empty array on that case, and the loading state would be changed by `setSearchedDashboardIdList`.
This issue caused problems on cost analysis page before, and that made users think that's kind of bugs, so we discussed about that problem and fixed together.